### PR TITLE
chore(release): 4.0.0-rc.0

### PR DESCRIPTION
**Pull request by evilebottnawi**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case

rc release

### Breaking Changes

No

### Additional Info

/cc @hiroppy @Loonride please check my changelog, maybe I forgot something

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,42 @@
...
+### Bug Fixes
+
+* respect `output.path` and `output.publicPath` options from the configuration
+* respect the `stats` option from the configuration
+* respect the `watchOptions` option from the configuration
+* the `writeToDisk` option now correctly works in multi-compiler mode
+* the `outputFileSystem` option now correctly works in multi-compiler mode
+* respect `[hash]`/`[fullhash]` in `output.path` and `output.publicPath`
+* handle exceptions for filesystem operations
+* the `Content-Type` header doesn't has `charset=utf-8` value for custom MIME types and MIME types which can be non `utf-8`
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the `Content-Type` header doesn't have `charset=utf-8` value for custom MIME types and MIME types which can be non `utf-8`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,42 @@
...
+
+* validate options
+* migrate on the `webpack` logger
+* migrate on the `memfs` package
+* improve performance
+
+### BREAKING CHANGES
+
+* minimum supported Node.js version is `10.13.0`
+* the default value of the option `publicPath` takes from the value of the `output.publicPath` option from the configuration (`webpack.config.js`)
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the default value of the option `publicPath` is taken from the value of the `output.publicPath` option from the configuration (`webpack.config.js`)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,42 @@
...
+* the `stats` option was removed, the default value of the `stats` option takes from the value of the `stats` option from the configuration (`webpack.config.js`)
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the `stats` option was removed, the default value of the `stats` option is taken from the value of the `stats` option from the configuration (`webpack.config.js`)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,42 @@
...
+* the `watchOptions` was removed, the default value of the `watchOptions` option takes from the value of the `watchOptions` option from the configuration (`webpack.config.js`)
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the `watchOptions` was removed, the default value of the `watchOptions` option is taken from the value of the `watchOptions` option from the configuration (`webpack.config.js`)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,42 @@
...
+* the `Content-Type` header doesn't has `charset=utf-8` value for custom MIME types and MIME types which can be non `utf-8`
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the `Content-Type` header doesn't have `charset=utf-8` value for custom MIME types and MIME types which can be non `utf-8`
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,42 @@
...
+* the `fs` option was renamed to the `outputFileSystem` option
+* the `lazy` option was removed without replacement
+* the `logger`, `logLevel` and `logTime` options were removed without replacement. You can setup the `level` value using `{ infrastructureLogging: { level: 'warn' } }`, please read https://webpack.js.org/configuration/other-options/#infrastructurelogging. You can use the `infrastructurelog` (`infrastructureLog` in `webpack@5`) hook to customize logs. The `log` property in the middleware context was renamed to `logger`
+* the `mimeTypes` option first requires you to specify an extension and then a content-type - `{ mimeTypes: { phtml: 'text/html' } }`
+* the `force` option from the `mimeTypes` option was removed without replacement 
+* the `reporter` option was removed without replacement
+* the `getFilenameFromUrl` method was removed from API
```
Comment (jamesgeorge007) [inline CHANGELOG.md]:
```suggestion
* the `getFilenameFromUrl` method was removed from the API
```

**Comment by evilebottnawi** [comment]
@jamesgeorge007 big thanks!

**Pull request merged**